diagrams/app_diagrams_generator.rb

```diff
@@ -0,0 +1,219 @@
+require 'yaml'
+require 'fileutils'
+
+class AppDiagramGenerator
```

Comment:
There's a lot going on in this class, it would be useful to have a test that exercises it all, to avoid it from breaking in the future.

Reply:
added some tests